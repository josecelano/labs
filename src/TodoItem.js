/**
 * This component should render a todo item as an <input type="checkbox" /> with a label
 */
import React from 'react';

const TodoItem = (props) => (
    <label>
        <input
            type="checkbox"
            checked={props.checked}
            onChange={() => props.onClick(props.text)}
            onClick={() => props.onClick(props.text)}/> {props.text}
    </label>
)

export default TodoItem;