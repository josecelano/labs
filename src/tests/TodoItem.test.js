/**
 * Tests for the TodoItem
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TodoItem from '../TodoItem';

describe('<TodoItem />', () => {
    it('should render an item', () => {
        const tree = renderer
            .create(<TodoItem />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should attach an onclick handler passed in', () => {
        const onClickSpy = jest.fn();
        const todoItem = shallow(
            <TodoItem onClick={onClickSpy}/>
        );
        todoItem.find('input[type="checkbox"]').simulate('change');
        expect(onClickSpy).toHaveBeenCalled();
    })
})