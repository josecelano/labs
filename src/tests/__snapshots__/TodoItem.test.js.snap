exports[`test should render a todo item as a checkbox with a label 1`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox" />
   
  Get to Las Palmas ðŸŒ´
</label>
`;
